### ISSUE TITLE
specifying multiple properties on MSBuild task only adds the last property to the executed command

### ISSUE THREAD

**Issue by Philo** [cake-build/cake#680]
See [The test branch](https://github.com/Philo/example/blob/issue/multiple-props-msbuild/build.cake#L39) on the example repository for an example.

Ensure you increase the buffer size of your command window to 9999, run the build:

`./build.ps1`

then scroll back to the start of the msbuild output:

![image](https://cloud.githubusercontent.com/assets/432217/12921821/dd964134-cf46-11e5-97ad-75b76cfcf0cc.png)

from build task:

```c#
      MSBuild("./src/Example.sln", settings => settings
		.SetConfiguration(configuration)
		.WithProperty("firstProperty", "one")
		.WithProperty("secondProperty", "two")
		.SetVerbosity(Verbosity.Diagnostic));
```

Looking at the unit tests this appears to be covered by passing tests (Should_Append_Property_With_Multiple_Values_To_Process_Arguments) yet I'm not seeing this behaviour.

**Comment by patriksvensson** [comment]
@Philo I'm unable to reproduce this. Are you sure that you built the right branch?

![image](https://cloud.githubusercontent.com/assets/357872/12941975/b96f9a8a-cfd7-11e5-9a5a-3cc1281704a6.png)

**Comment by devlead** [comment]
@Philo this is indeed weird issue, can't see in the code how it would occur in the code and like @patriksvensson I currently can't reproduce it.

I just cloned the repo, switched branch and ran build.ps1 and filtered log for those outputs and it's both in log and args.
![image](https://cloud.githubusercontent.com/assets/1647294/12942192/4d0abda0-cfd9-11e5-855e-07d784875b40.png)

**Comment by Philo** [comment]
Indeed, I'm inclined to think this could even be an issue with the diagnostic reporting from MSBuild itself, as I was able to get my scenario working (running MSBuild with the "Package" target for generating a web deploy zip.  This requires that MSBuild is building the web csproj file and if you try to set a PlatformTarget you get an MSBuild error that was originally assuming was related to this.

Happy to close this issue as not actually a cake issue.

NB: My log output from from MSBuild and not the cake diagnostic logging.

**Comment by devlead** [comment]
@Philo you're indeed right, now see that we have the same issue in our screen shots, we should file an issue to @Microsoft/msbuild if one hasn't already been made.
The property gets set and I can use it in the MSBuild scripts, so hopefully it's just an display error.

**Comment by devlead** [comment]
@Philo FYI and fix for this was just merged in Microsoft/msbuild#477
Kudos to you for finding it :+1:

**Comment by patriksvensson** [comment]
@devlead Great job fixing this!
